### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by aaprasad**
Here we address #54 by adding tutorials, and more documentation

## Summary by CodeRabbit

- **Documentation**
  - Enhanced README with detailed setup, training, and inference instructions.
  - Added comprehensive parameter guides for training and inference configurations.
  - Introduced detailed documentation for new models and modules, including `GlobalTrackingTransformer` and `GTRRunner`.

- **New Features**
  - Added support for batch inference with multiple models.
  - Introduced new configuration files for inference and training.
  
- **Style**
  - Improved documentation styling with new CSS for better readability.

- **Bug Fixes**
  - Corrected docstring indentation for better clarity.
  
- **Refactor**
  - Updated type annotations across various methods for better type checking and readability.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,4 +19,117 @@ conda env create -y -f environment.yml && conda activate biogtr
...
+        vid_1.{LABELS_EXTENSION}
+        ...
+        vid_n.{VID_EXTENSION}
+        vid_n.{LABELS_EXTENSION}
+    inference/
+        vid_1.{VID_EXTENSION}
+        vid_1.{LABELS_EXTENSION} # these files don't need tracks
+        ...
+        vid_n.{VID_EXTENSION}
+        vid_n.{LABELS_EXTENSION}
```
Comment (talmo) [inline README.md:70]: Put these in another section when you get to inference.

Also, do we really support any labels extension that isn't `.slp`? Either way, let's just be explicit.

We should also be explicit about the video extensions. Especially since we're doing microscopy, it would be good to know if zarr and multi-frame TIFF stacks are supported. Since `sleap-io` provides the layer for doing video reading, it's totally okay to redirect users to the [sleap-io docs](https://io.sleap.ai/latest/formats/#sleap_io.load_video). (I'll add more explicit info there.)

File: biogtr/training/configs/README.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
+# Description of training parameters
```
Comment (talmo) [inline biogtr/training/configs/README.md:1]: See above -- needs link to a concrete example.

File: biogtr/training/configs/README.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+```YAML
+augmentations: 
+    Rotate:
+        limit: 45
+        p: 0.3
+    ...
+    MotionBlur:
+        blur_limit: [3,7]
+        p: 0.3
+```
```
Comment (talmo) [inline biogtr/training/configs/README.md:142]: This is great! A concrete example for how this subsection should be written. Have more of these everywhere.

File: biogtr/training/configs/README.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+### `dataloader`
+
+This section outlines the params needed for the dataloader. Should have a `train_dataloader` and optionally `val_dataloader`/`test_dataloader` keys. See [`torch.utils.data.Dataloader`](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) for available args.
```
Comment (talmo) [inline biogtr/training/configs/README.md:146]: Provide example and you NEED an entry for each field that we actually use in the yaml.

The docs you link to take many arguments that are not YAML-serializable (e.g., function handles), and it's also very unclear how these args are used in this codebase specifically.

File: biogtr/training/configs/README.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+### `logging`:
+
+This section sets up logging for the training job. 
+
+* `logger_type`: A string indicating which logger to use. Available loggers are {`"CSVLogger"`, `"TensorBoardLogger"`,`"WandbLogger"`}
+
+Note: Please see the documentation for the corresponding logger at [`lightning.loggers`](https://lightning.ai/docs/pytorch/stable/api_references.html#loggers) for available parameters.
```
Comment (talmo) [inline biogtr/training/configs/README.md:154]: No. Include the documentation HERE.

**Review by talmo — CHANGES_REQUESTED** [review]

**Review by talmo — APPROVED** [review]
